src/gui/qgslayermetadataresultsmodel.cpp

```diff
@@ -0,0 +1,216 @@
+/***************************************************************************
+  qgslayermetadataresultsmodel.cpp - QgsLayerMetadataResultsModel
+
+ ---------------------
+ begin                : 1.9.2022
+ copyright            : (C) 2022 by ale
+ email                : [your-email-here]
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgslayermetadataresultsmodel.h"
+#include "qgsfeedback.h"
+#include "qgsapplication.h"
+#include "qgslayermetadataproviderregistry.h"
+#include "qgslayermetadataformatter.h"
+#include <QIcon>
+
+QgsLayerMetadataResultsModel::QgsLayerMetadataResultsModel( const QgsMetadataSearchContext &searchContext, QObject *parent )
+  : QAbstractTableModel( parent )
+  , mSearchContext( searchContext )
+{
+  qRegisterMetaType< QgsLayerMetadataSearchResults>( "QgsLayerMetadataSearchResults" );
+  qRegisterMetaType< QgsLayerMetadataProviderResult>( "QgsLayerMetadataProviderResult" );
+}
+
+QgsLayerMetadataResultsModel::~QgsLayerMetadataResultsModel()
+{
+  cancel();
+}
+
+int QgsLayerMetadataResultsModel::rowCount( const QModelIndex &parent ) const
+{
+  return parent.isValid() ? 0 : mResult.metadata().count();
+}
+
+int QgsLayerMetadataResultsModel::columnCount( const QModelIndex &parent ) const
+{
+  return parent.isValid() ? 0 : 5;
+}
+
+QVariant QgsLayerMetadataResultsModel::data( const QModelIndex &index, int role ) const
+{
+  if ( index.isValid() && index.row() < mResult.metadata().count( ) )
+  {
+    switch ( role )
+    {
+      case Qt::ItemDataRole::DisplayRole:
+      {
+        switch ( index.column() )
+        {
+          case Sections::Identifier:
+            return mResult.metadata().at( index.row() ).identifier( );
+          case Sections::Title:
+            return mResult.metadata().at( index.row() ).title();
+          case Sections::Abstract:
+            return mResult.metadata().at( index.row() ).abstract();
+          case Sections::DataProviderName:
+            return mResult.metadata().at( index.row() ).dataProviderName();
```

Comment:
Would be nicer to get the QgsProviderMetadata::description() associated with the provider -- that's a more user friendly name.